docs/src/pages/demos/badges/BadgeMax.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles, Theme, createStyles, WithStyles } from '@material-ui/core/styles';
+import Badge from '@material-ui/core/Badge';
+import MailIcon from '@material-ui/icons/Mail';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    margin: {
+      margin: theme.spacing(2),
+      marginRight: theme.spacing(3),
+    },
+  });
+
+function BadgeMax(props: WithStyles<typeof styles>) {
+  const { classes } = props;
+
+  return (
+    <React.Fragment>
+      <Badge className={classes.margin} badgeContent={99} color="primary">
+        <MailIcon />
+      </Badge>
+      <Badge className={classes.margin} badgeContent={100} color="primary">
+        <MailIcon />
+      </Badge>
+      <Badge className={classes.margin} badgeContent={1000} max={999} color="primary">
+        <MailIcon />
+      </Badge>
+    </React.Fragment>
+  );
+}
+
+BadgeMax.propTypes = {
+  classes: PropTypes.object.isRequired,
+} as any;
+
+export default withStyles(styles)(BadgeMax);
```

Comment:
@eps1lon Are you sure you don't want to do the TypeScript and hoc to hook migration at the same time? I'm wondering if it won't make the process more efficient.

Reply:
Draft a proposal with the process. Otherwise I don't understand what changed since the last discussion.

Reply:
We have a bit more hindsight, I was interested in your point of view. We can change our mind. It happens to me all the time.